sympy/integrals/intpoly.py

```diff
@@ -0,0 +1,448 @@
+from __future__ import print_function, division
+
+from sympy.core import S, Symbol
+from sympy.geometry.polygon import Polygon
+from sympy.geometry.point import Point
+from sympy.core.function import diff
+from sympy.core.expr import Expr
+from sympy.abc import x, y
+from sympy.polys.polytools import gcd_list
+
+
+def polytope_integrate(poly, expr, dims=None):
+    """This is currently a basic prototype for integrating
+    univariate/bivariate polynomials over 2-Polytopes.
+    Parameters
+    ==========
+    poly : The input Polygon.
+    expr : The input polynomial.
+    dims : The tuple of symbols denoting axes.
+    Example
+    =======
+    >>> from sympy.abc import x, y
+    >>> from sympy.geometry.polygon import Polygon
+    >>> from sympy.geometry.point import Point
+    >>> from sympy.integrals.intpoly import polytope_integrate
+    >>> poly = Polygon(Point(0,0), Point(0,1), Point(1,1), Point(1,0))
+    >>> expr = x*y
+    >>> polytope_integrate(poly, expr)
+    1/4
+    """
+    if dims is None:
+        if isinstance(expr, Expr):
+            dims = list(expr.free_symbols)
```

Comment:
Can we have `tuple(expr.free_symbols)`? I don't think you are using mutability of `dims`.